src/contributing/prs.md

```diff
@@ -0,0 +1,150 @@
+     this make something easier or harder in the future?). Documentation comes
+     in the form of comments in code, “external” documentation (such as you
+     read here), and commit and pull request messages. In whatever form it
+     comes, documentation must be clear (as easy to understand as possible),
+     concise (as short as possible while getting all the required points
+     across), and complete (not missing any important points).
+
+  4. *Useful*. The more people that will benefit from a pull request, the more
+     likely it is to be merged. The bar for “useful” is set fairly low, but
+     that does not mean that every pull request should be merged: for example,
+     a pull request which simply changes code to your preferred style is not
+     only not useful but may cause problems for other people working in
+     parallel. However, porting the code to a little used platform probably is
+     often useful (provided it doesn’t cause undue problems for more widely
+     used platforms).
+
+
+## The pull request process in yk repositories
+
+To raise a pull request, you must:
+
+  1. Fork the relevant yk GitHub repository to your account and clone it to
+     your local computer. Note that pull requests must never come from a branch
+     on a yk repository.
+
+  2. Create a new branch in your fork. If you are unsure about the name, start
+     with something generic and rename it later with `git branch -m <new
+     name>`.
+
+  3. Make your changes and commit them. It is not only allowed, but often best,
+     to have multiple commits: each commit should be a logical change building
+     upon one of its predecessors. Each commit should be capable of passing all
+     tests successfully (amongst other things, to avoid breaking `git bisect`).
+
+  4. Push your branch to your GitHub fork and raise a pull request. Give the
+     pull request a descriptive name (for example, “Fix bug” is not useful but
+     “Deal with an empty list correctly” is) and a description (empty
+     descriptions are almost never acceptable). If it fixes a GitHub issue, add
+     the text `Fixes #<GitHub issue number>` into the description.
+
+Your pull request has now been raised and will be reviewed by a yk contributor.
+The aim of a review is two fold: to ensure that contributions to yk are of an
+acceptable quality; and that at least two people (the reviewer and reviewee)
+understand what was changed and why.
+
+The reviewer will comment in detail on specific lines of the commit and on the
+overall pull request. Comments might be queries (e.g. “can the list ever be
+empty at this point?”) or corrections (e.g. “we need to deal with the
+possibility of an empty list at this point”). For each comment you can either:
+
+  1. Address the issue. For example, if the reviewer has pointed out correctly
+     that something needs to be changed, make a small additional commit, push
+     it to your branch and make a comment `Fixed in #<git hash>` so that the
```

Comment:
I've never put a hash before a commit ID. Typo?

Reply:
Fixed in 71eb1e1.